src/abstractdataframe/selection.jl

```diff
@@ -6,6 +6,18 @@
 # 4) Pair{AsTable, <:Pair{<:Base.Callable, <:Union{Symbol, Vector{Symbol}, Type{AsTable}}}}
 # 5) Callable
 
+const TRANSFORM_METADATA = """
+Metadata: this function propagates table level metadata that has :note style.
+Column level metadata that has :note style is propagated if:
+a) a single column is transformed to a single column and the name of the column
+does not change (this includes all column selection operations), or
+b) a single column is transformed with `identity` or `copy` to a single column
+   even if column name is changed (this includes column renaming).
+   As a special case for `GroupedDataFrame` if this transformation is made
+   into a grouping column that is kept it is ignored (metadata of grouping
+   column is retained).
```

Comment:
if you do:
```
gdf = groupby(df, :a)
combine(gdf, :b => :a)
```
this operation succeeds if `:b` column had the same values as `:a` column. However, `:b` column could have different metadata than `:a` column. In this case metadata of `:a` column is kept in `:a` column, in other words metadata of `:b` column is ignored.

However, if you do:
```
gdf = groupby(df, :a)
combine(gdf, :b => :a, keepkeys=false)
```
then metadata from `:b` column is stored in `:a` column (and - but this is not part of this PR, `:b` column can store any values).

Reply:
Wow, that's really a corner case of a corner case. :-D Does this exception exist because it's easier to implement it that way?

Maybe say it like this?
```suggestion
   even if column name is changed (this includes column renaming).
   As a special case for `GroupedDataFrame` if the output replaces
   a grouping column and `keepkeys=true`, metadata is taken from
   original grouping column.
```

Reply:
> Does this exception exist because it's easier to implement it that way?

The exception exists because technically we keep the grouping column. So, for example, if it has some special vector type, and column `:b` would have other type (but the same values) this special vector type would be kept:

```
julia> df = DataFrame(a=PooledArray(1:3), b=1:3)
3×2 DataFrame
 Row │ a      b
     │ Int64  Int64
─────┼──────────────
   1 │     1      1
   2 │     2      2
   3 │     3      3

julia> gdf = groupby(df, :a)
GroupedDataFrame with 3 groups based on key: a
First Group (1 row): a = 1
 Row │ a      b
     │ Int64  Int64
─────┼──────────────
   1 │     1      1
⋮
Last Group (1 row): a = 3
 Row │ a      b
     │ Int64  Int64
─────┼──────────────
   1 │     3      3

julia> res1 = combine(gdf, :b => :a)
3×1 DataFrame
 Row │ a
     │ Int64
─────┼───────
   1 │     1
   2 │     2
   3 │     3

julia> res1.a
3-element PooledVector{Int64, UInt32, Vector{UInt32}}:
 1
 2
 3

julia> res2 = combine(gdf, :b => :a, keepkeys=false)
3×1 DataFrame
 Row │ a
     │ Int64
─────┼───────
   1 │     1
   2 │     2
   3 │     3

julia> res2.a
3-element Vector{Int64}:
 1
 2
 3
```

Reply:
OK. It could be more logical to use the new column instead, given that the user explicitly requested that, but that's not a big deal (and it's not mainly a metadata issue).

Maybe "replace" wasn't the right term in my suggestion then. How about this?
```
   even if column name is changed (this includes column renaming).
   As a special case for `GroupedDataFrame` if the output has the same name
   as a grouping column and `keepkeys=true`, metadata is taken from
   original grouping column.
```

Reply:
The reason why we keep grouping column is that it is likely to be optimized for grouping (like in my example - `PooledArray` vs `Vector`) + this is a super old behavior (pre-my time) and changing it would be breaking.

Docstring is fixed.